src/components/modals/ManageProjectAddresses/AddNewAddress.tsx

```diff
@@ -65,26 +74,26 @@ export const AddNewAddress: FC<IAddNewAddress> = ({
 							{
 								address: _address,
 								isRecipient: true,
-								networkId: selectedWallet.networkId,
+								networkId: selectedChain.id,
+								chainType,
 							},
 						];
 						newProjects.push(structuredClone(_project));
 					} else {
 						newProjects.push(_project);
 					}
 				}
-				setSelectedWallet(undefined);
+				setSelectedChain(undefined);
 				return newProjects;
 			});
 			setAddresses(_addresses => {
 				const _adds = structuredClone(_addresses);
-				for (let i = 0; i < _adds.length; i++) {
-					const _add = _adds[i];
-					if (_add.networkId === selectedWallet.networkId) {
-						_add.isRecipient = true;
-						_add.address = _address;
-					}
-				}
+				_adds.push({
+					address: _address,
+					isRecipient: true,
+					networkId: selectedChain.id,
+					chainType,
+				});
```

Comment:
I don't understand why this new logic works! Was there any issue with the older implementation? Because the new one is pushing a new item to the list while the previous one wasn't!

Reply:
older implementation was using an array of wallet addresses and mapping that array to create "AddNewAddress" instances. It was confusing because not all projects have addresses in all chains. It's more reasonable to map Chains instead of wallet addresses. like in create project page and manage funds in verification view, we use Chains to map, so it's better to use Chains here as well instead of mutating the original wallet addresses array, and then we'll have a consistent code along the app.

Besides that, with this new logic, I'm using the original addresses that we fetch from backend and I'm not mutating it, that is the best practices. In older implementation, we were using a useEffect to mutate original addresses array, that is not necessary . So, now, when user adds a new address to a new chain, I call backend to update project addresses then use "push" to update front-end states as well.

Reply:
@MohammadPCh to weigh in here.

Reply:
Guys thank you for this conversation,
@RamRamez about the first part, I don't agree with you about the confusing part. these are two different approaches and each of them has its benefits. 
and if we gonna talk about the second part, I do not agree with you about the best practices. we have a state and update it when the user adds something.
@jainkrati

Reply:
Thanks @MohammadPCh. I disagree with you about both parts.

1. Could you please explain the benefits of older implementation?
You added a useEffect that applies unnecessary rerender by updating "addresses" state. We don't need to add an empty field for all chains in addresses array because user may not fill them all. You could simply replace "addresses.map" with "CHAINS.map" and remove the big useEffect that causes rerenders and unnecessary computations.

2. Could you please explain the bugs you have found in my codes?

Reply:
@RamRamez, thank you for your response. 
Please note that my comment pertains to the subject Amin brought up in the call – that we should avoid refactoring when we're under pressure to deliver – and I confirmed his statement.
Regarding your first question, the previous approach was straightforward. I'm not sure why you found it confusing. We maintained an array of all chains and displayed them in the modal. When a user added an address, we updated its corresponding element. It's quite close to your idea, and we didn't reject it.
Generally, I don't check all the lines, I just take a quick look at them and check best practices.
Please be aware that when a 'useEffect' only runs once, it doesn't have a detrimental impact on code performance.
For the second question, could you please revisit my comment? Why do you think that way? 😄 
I want to emphasize once again that this discussion revolves around avoiding unnecessary changes that make the PR larger and more challenging to identify potential issues in the future.
@jainkrati

Reply:
Thanks @MohammadPCh . Only 100 lines has changed and I can say it's a small PR.
But, after two weeks this PR is not yet merged. @MohammadPCh I'm really confused, have you reviewed this PR? Is it on approved state or requested changes state?

Reply:
@RamRamez, as I mentioned, I haven't reviewed this PR yet because it seems unnecessary, given that you've already received approval from both Amin and Mohammad. Krati asked me to share my opinion on your discussions with Amin, and I have provided my feedback.

Reply:
For reducing risk of regression, lets revert the unnecessary changes @RamRamez